From 3e19d281053e24ed67a7884f622d685aec8a85b4 Mon Sep 17 00:00:00 2001
From: "Padmasanan, Prathisna" <prathisna.padmasanan@intel.com>
Date: Wed, 21 Apr 2021 14:49:51 +0530
Subject: [PATCH 034/109] Fix memory allocation for hddl client data

memory allocated in device memory using devm_kcalloc was freed using
kfree() that lead to memory corruption.Memory allocation using kcalloc()
fixes the issue.

Signed-off-by: Padmasanan, Prathisna <prathisna.padmasanan@intel.com>
---
 drivers/misc/hddl_device/hddl_device.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/hddl_device/hddl_device.c b/drivers/misc/hddl_device/hddl_device.c
index 8571f70f0361..0a28406afa6f 100644
--- a/drivers/misc/hddl_device/hddl_device.c
+++ b/drivers/misc/hddl_device/hddl_device.c
@@ -174,9 +174,9 @@ struct intel_hddl_clients **
 		if (!cls)
 			return NULL;
 	} else {
-		cls = devm_kcalloc(dev, num_devices,
-				   sizeof(struct intel_hddl_clients *),
-				   GFP_KERNEL);
+		cls = kcalloc(num_devices,
+			      sizeof(struct intel_hddl_clients *),
+			      GFP_KERNEL);
 		if (!cls)
 			return NULL;
 		/*
-- 
2.25.1

